--- NuSMV-2.5.4/cudd-2.4.1.1/util/pipefork.c.orig	2016-10-18 19:09:45.873681378 +0200
+++ NuSMV-2.5.4/cudd-2.4.1.1/util/pipefork.c	2016-10-18 19:11:50.617472653 +0200
@@ -12,6 +12,11 @@
 #include <sys/wait.h>
 #endif
 
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+#include <sys/wait.h>
+#include <unistd.h>
 
 #ifndef __GNUC__
 extern pid_t wait3 ARGS((int *statusp, int options, struct rusage *rusage));
@@ -40,7 +45,7 @@
     int topipe[2], frompipe[2];
     char buffer[1024];
 
-#if (defined __hpux) || (defined __osf__) || (defined _IBMR2) || (defined __SVR4) || (defined __CYGWIN32__) || (defined __MINGW32__) 
+#if (defined __hpux) || (defined __osf__) || (defined _IBMR2) || (defined __SVR4) || (defined __CYGWIN32__) || (defined __MINGW32__) || defined(__linux__)
     int status;
 #else
     union wait status;
