--- NuSMV-2.4.0/nusmv/doc/tutorial/Makefile.in	2005-05-09 10:30:18.000000000 +0200
+++ NuSMV-2.4.0-ok/nusmv/doc/tutorial/Makefile.in	2006-10-03 17:03:50.000000000 +0200
@@ -470,9 +470,9 @@
 @HAVE_PDFLATEXETAL_FALSE@		&& latex main.tex
 
 @HAVE_PDFLATEXETAL_TRUE@install-data-local: tutorial.ps
-@HAVE_PDFLATEXETAL_TRUE@	$(install_sh_DATA) tutorial.ps $(datadir)/nusmv/doc/tutorial.ps
+@HAVE_PDFLATEXETAL_TRUE@	$(install_sh_DATA) tutorial.ps $(DESTDIR)$(datadir)/nusmv/doc/tutorial.ps
 
 @HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@install-data-local: tutorial.ps
-@HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@	$(install_sh_DATA) tutorial.ps $(datadir)/nusmv/doc/tutorial.ps
+@HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@	$(install_sh_DATA) tutorial.ps $(DESTDIR)$(datadir)/nusmv/doc/tutorial.ps
 
 @HAVE_LATEX_FALSE@@HAVE_PDFLATEXETAL_FALSE@install-data-local:
 @HAVE_LATEX_FALSE@@HAVE_PDFLATEXETAL_FALSE@	@echo Cannot generate tutorial: latex not found
--- NuSMV-2.4.0/nusmv/doc/user-man/Makefile.in	2005-10-26 11:11:25.000000000 +0200
+++ NuSMV-2.4.0-ok/nusmv/doc/user-man/Makefile.in	2006-10-03 17:04:09.000000000 +0200
@@ -489,9 +489,9 @@
 @HAVE_PDFLATEXETAL_FALSE@		&& latex main.tex
 
 @HAVE_PDFLATEXETAL_TRUE@install-data-local: nusmv.ps
-@HAVE_PDFLATEXETAL_TRUE@	$(install_sh_DATA) nusmv.ps $(datadir)/nusmv/doc/nusmv.ps
+@HAVE_PDFLATEXETAL_TRUE@	$(install_sh_DATA) nusmv.ps $(DESTDIR)$(datadir)/nusmv/doc/nusmv.ps
 @HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@install-data-local: nusmv.ps
-@HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@	$(install_sh_DATA) nusmv.ps $(datadir)/nusmv/doc/nusmv.ps
+@HAVE_LATEX_TRUE@@HAVE_PDFLATEXETAL_FALSE@	$(DESTDIR)$(install_sh_DATA) nusmv.ps $(datadir)/nusmv/doc/nusmv.ps
 @HAVE_LATEX_FALSE@@HAVE_PDFLATEXETAL_FALSE@install-data-local:
 @HAVE_LATEX_FALSE@@HAVE_PDFLATEXETAL_FALSE@	@echo Cannot generate manual: latex not found
 
diff -ur NuSMV-2.4.0/nusmv/helpers/extract_doc.in NuSMV-2.4.0-ok/nusmv/helpers/extract_doc.in
--- NuSMV-2.4.0/nusmv/helpers/extract_doc.in	2005-03-03 13:12:59.000000000 +0100
+++ NuSMV-2.4.0-ok/nusmv/helpers/extract_doc.in	2006-10-03 17:29:10.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # FileName	[ extract_doc.in ]
 #
--- NuSMV-2.4.3/cudd-2.4.1.1/Makefile.orig	2007-05-10 19:37:13.000000000 +0200
+++ NuSMV-2.4.3/cudd-2.4.1.1/Makefile	2007-09-14 15:33:55.000000000 +0200
@@ -55,7 +55,7 @@
 # These four are typical settings for optimized code with gcc.  The
 # last two also work with icc/ecc.
 #ICFLAGS	= -g -O6 -Wall
-ICFLAGS	= -g -O6
+#ICFLAGS	= -g -O6
 #ICFLAGS	= -g -O3 -Wall
 #ICFLAGS	= -g -O3
 
@@ -69,7 +69,7 @@
 # Gcc 2.8.1 or higher on i686.
 #XCFLAGS	= -mcpu=pentiumpro -malign-double -DHAVE_IEEE_754 -DBSD
 # Gcc 3.2.2 or higher on i686.
-XCFLAGS	= -mcpu=pentium4 -malign-double -DHAVE_IEEE_754 -DBSD -DSIZEOF_VOID_P=4 -DSIZEOF_LONG=4 -DSIZEOF_INT=4
+XCFLAGS	= -malign-double -DHAVE_IEEE_754 -DBSD -DSIZEOF_VOID_P=4 -DSIZEOF_LONG=4 -DSIZEOF_INT=4
 # Icc on i686.
 #XCFLAGS	= -ansi -align -ip -DHAVE_IEEE_754 -DBSD
 # Gcc on ia64.
--- NuSMV-2.4.3/cudd-2.4.1.1/Makefile_64bit~	2007-03-14 11:31:56.000000000 +0100
+++ NuSMV-2.4.3/cudd-2.4.1.1/Makefile_64bit	2007-09-14 15:35:25.000000000 +0200
@@ -55,7 +55,7 @@
 # These four are typical settings for optimized code with gcc.  The
 # last two also work with icc/ecc.
 #ICFLAGS	= -g -O6 -Wall
-ICFLAGS	= -g -O6
+#ICFLAGS	= -g -O6
 #ICFLAGS	= -g -O3 -Wall
 #ICFLAGS	= -g -O3
 
