diff -ur NuSMV-2.4.0/nusmv/doc/html/Makefile.am NuSMV-2.4.0-ok/nusmv/doc/html/Makefile.am
--- NuSMV-2.4.0/nusmv/doc/html/Makefile.am	2005-03-03 12:05:28.000000000 +0100
+++ NuSMV-2.4.0-ok/nusmv/doc/html/Makefile.am	2006-10-03 17:30:35.000000000 +0200
@@ -46,7 +46,7 @@
 docs: helpfiles
 
 helpfiles: 
-	cd $(top_builddir) && $(SHELL) helpers/extract_doc .
+	cd $(top_builddir) && /bin/bash helpers/extract_doc .
 
 install: docs install-am
 
diff -ur NuSMV-2.4.0/nusmv/doc/tutorial/Makefile.am NuSMV-2.4.0-ok/nusmv/doc/tutorial/Makefile.am
--- NuSMV-2.4.0/nusmv/doc/tutorial/Makefile.am	2005-05-09 10:30:18.000000000 +0200
+++ NuSMV-2.4.0-ok/nusmv/doc/tutorial/Makefile.am	2006-10-03 17:03:50.000000000 +0200
@@ -62,7 +62,7 @@
 
 if HAVE_LATEX
 install-data-local: tutorial.ps
-	$(install_sh_DATA) tutorial.ps $(datadir)/nusmv/doc/tutorial.ps
+	$(install_sh_DATA) tutorial.ps $(DESTDIR)$(datadir)/nusmv/doc/tutorial.ps
 
 else 
 install-data-local:
diff -ur NuSMV-2.4.0/nusmv/doc/user-man/Makefile.am NuSMV-2.4.0-ok/nusmv/doc/user-man/Makefile.am
--- NuSMV-2.4.0/nusmv/doc/user-man/Makefile.am	2005-10-26 11:11:25.000000000 +0200
+++ NuSMV-2.4.0-ok/nusmv/doc/user-man/Makefile.am	2006-10-03 17:04:09.000000000 +0200
@@ -67,7 +67,7 @@
 
 if HAVE_LATEX
 install-data-local: nusmv.ps
-	$(install_sh_DATA) nusmv.ps $(datadir)/nusmv/doc/nusmv.ps
+	$(install_sh_DATA) nusmv.ps $(DESTDIR)$(datadir)/nusmv/doc/nusmv.ps
 else 
 install-data-local:
 	@echo Cannot generate manual: latex not found
diff -ur NuSMV-2.4.0/nusmv/helpers/extract_doc.in NuSMV-2.4.0-ok/nusmv/helpers/extract_doc.in
--- NuSMV-2.4.0/nusmv/helpers/extract_doc.in	2005-03-03 13:12:59.000000000 +0100
+++ NuSMV-2.4.0-ok/nusmv/helpers/extract_doc.in	2006-10-03 17:29:10.000000000 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 # FileName	[ extract_doc.in ]
 #
--- NuSMV-2.4.3/cudd-2.4.1.0/Makefile.orig	2007-05-10 19:37:13.000000000 +0200
+++ NuSMV-2.4.3/cudd-2.4.1.0/Makefile	2007-09-14 15:33:55.000000000 +0200
@@ -55,7 +55,7 @@
 # These four are typical settings for optimized code with gcc.  The
 # last two also work with icc/ecc.
 #ICFLAGS	= -g -O6 -Wall
-ICFLAGS	= -g -O6
+#ICFLAGS	= -g -O6
 #ICFLAGS	= -g -O3 -Wall
 #ICFLAGS	= -g -O3
 
@@ -69,7 +69,7 @@
 # Gcc 2.8.1 or higher on i686.
 #XCFLAGS	= -mcpu=pentiumpro -malign-double -DHAVE_IEEE_754 -DBSD
 # Gcc 3.2.2 or higher on i686.
-XCFLAGS	= -mcpu=pentium4 -malign-double -DHAVE_IEEE_754 -DBSD -DSIZEOF_VOID_P=4 -DSIZEOF_LONG=4 -DSIZEOF_INT=4
+XCFLAGS	= -malign-double -DHAVE_IEEE_754 -DBSD -DSIZEOF_VOID_P=4 -DSIZEOF_LONG=4 -DSIZEOF_INT=4
 # Icc on i686.
 #XCFLAGS	= -ansi -align -ip -DHAVE_IEEE_754 -DBSD
 # Gcc on ia64.
--- NuSMV-2.4.3/cudd-2.4.1.0/Makefile_64bit~	2007-03-14 11:31:56.000000000 +0100
+++ NuSMV-2.4.3/cudd-2.4.1.0/Makefile_64bit	2007-09-14 15:35:25.000000000 +0200
@@ -55,7 +55,7 @@
 # These four are typical settings for optimized code with gcc.  The
 # last two also work with icc/ecc.
 #ICFLAGS	= -g -O6 -Wall
-ICFLAGS	= -g -O6
+#ICFLAGS	= -g -O6
 #ICFLAGS	= -g -O3 -Wall
 #ICFLAGS	= -g -O3
 
--- NuSMV-2.4.3/nusmv/src/Makefile.am~	2006-06-28 18:16:16.000000000 +0200
+++ NuSMV-2.4.3/nusmv/src/Makefile.am	2007-09-14 15:48:35.000000000 +0200
@@ -91,4 +91,6 @@
 	$(mbp_libs)              \
 	$(sa_libs)               \
 	$(mathsat_libs)          \
+	$(expat_libs)            \
+	$(readline_libs)         \
 	sm/libsm.la              
--- NuSMV-2.4.3/nusmv/Makefile.am~	2007-03-28 15:49:05.000000000 +0200
+++ NuSMV-2.4.3/nusmv/Makefile.am	2007-09-14 15:48:37.000000000 +0200
@@ -116,7 +116,7 @@
 libnusmv_la_LIBADD = $(top_builddir)/src/libnusmvcore.la \
 	-L$(top_srcdir)/$(CUDD_DIR)/lib \
 	-lcudd -lcudd_util -lmtr -lst   \
-        $(readline_libs) $(zchaff_libs) $(minisat_libs) $(expat_libs)
+        $(zchaff_libs) $(minisat_libs)
 
 libnusmv_la_DEPENDENCIES = $(top_builddir)/src/libnusmvcore.la $(CUDD_DEPENDENCY)
 
